environment-pipelines/tests/unit.tftest.hcl

```diff
@@ -392,6 +397,63 @@ run "test_code_pipeline" {
   }
 }
 
+run "test_pipeline_trigger_setup" {
+  command = plan
+
+  variables {
+    trigger_on_push = false
+  }
+
+  assert {
+    condition     = aws_codepipeline.environment_pipeline.stage[0].action[0].configuration.DetectChanges == "false"
+    error_message = "Should be: true"
+  }
+
+  assert {
+    condition     = contains(aws_codepipeline.environment_pipeline.trigger[0].git_configuration[0].push[0].branches[0].includes, "NO_TRIGGER")
+    error_message = "Should be: NO_TRIGGER"
+  }
+
+  assert {
+    condition     = length(aws_codepipeline.environment_pipeline.trigger[0].git_configuration[0].push[0].branches[0].includes) == 1
+    error_message = "Should be: true"
```

Comment:
```suggestion
    error_message = "Should be: 1"
```

Reply:
The length function evaluates this to true|false but I've added a better message anyway